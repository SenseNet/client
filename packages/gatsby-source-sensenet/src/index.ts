export * from './gatsby-node'
export * from './utils'
