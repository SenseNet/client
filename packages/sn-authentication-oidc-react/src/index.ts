import { UserManager, UserManagerSettings, WebStorageStateStore } from 'oidc-client'
export * from './components/authentication-provider'
export * from './components/oidc-secure'
export * from './use-authentication'
export { WebStorageStateStore, UserManagerSettings, UserManager }
