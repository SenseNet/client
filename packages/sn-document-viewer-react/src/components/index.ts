export * from './comment'
export * from './document-widgets'
export * from './page-widgets'
export * from './ConfirmationDialog'
export * from './DocumentViewer'
export * from './DocumentViewerError'
export * from './DocumentViewerLayout'
export * from './DocumentViewerLoading'
export * from './DocumentViewerRegeneratePreviews'
export * from './LayoutAppBar'
export * from './Page'
export * from './PageList'
export * from './styles'
