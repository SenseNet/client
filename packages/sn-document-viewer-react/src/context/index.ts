export * from './api-settings'
export * from './comment-states'
export * from './comments'
export * from './document-data'
export * from './document-permissions'
export * from './localization-context'
export * from './preview-image-data'
export * from './viewer-settings'
export * from './viewer-state'
