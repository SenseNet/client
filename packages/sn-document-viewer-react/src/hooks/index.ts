export * from './use-comment-state'
export * from './use-comments'
export * from './use-document-data'
export * from './use-document-permissions'
export * from './use-document-viewer-api'
export * from './use-localization'
export * from './use-preview-image'
export * from './use-preview-images'
export * from './use-viewer-settings'
export * from './use-viewer-state'
