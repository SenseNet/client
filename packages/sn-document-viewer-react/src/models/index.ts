export * from './DocumentViewerApiSettings'
export * from './Theming'
export * from './viewer-state'
