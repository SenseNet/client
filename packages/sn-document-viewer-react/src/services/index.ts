export * from './image-utils'
export * from './apply-shape-rotations'
