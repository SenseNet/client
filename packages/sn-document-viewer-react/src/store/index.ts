import { Action, applyMiddleware, compose, createStore, Store } from 'redux'
import { ReduxDiMiddleware } from 'redux-di-middleware'
import { DocumentViewerSettings } from '../models'
import { rootReducer, RootReducerType } from './RootReducer'

const composeEnhancers = (window as any).__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose

/**
 * gets a configuration object for the Store instance
 * @param {DocumentViewerSettings} settings The Settings object for the document viewer instance
 */
export const getStoreConfig = (settings: DocumentViewerSettings) => {
  const di = new ReduxDiMiddleware(settings.injector)
  di.setInjectable(settings)
  return {
    rootReducer,
    preloadedState: { sensenetDocumentViewer: { documentState: { isLoading: true } } },
    enhancer: composeEnhancers(applyMiddleware(di.getMiddleware())),
  }
}

/**
 * returns a Store object for a Document Viewer instance
 * @param {DocumentViewerSettings} settings The Settings object for the document viewer instance
 */
export const configureStore: (settings: DocumentViewerSettings) => Store<RootReducerType> = settings => {
  const config = getStoreConfig(settings)
  return createStore<RootReducerType, Action, {}, {}>(config.rootReducer, config.preloadedState, config.enhancer)
}

export * from './Comments'
export * from './Document'
export * from './PreviewImages'
export * from './RootReducer'
export * from './Viewer'
