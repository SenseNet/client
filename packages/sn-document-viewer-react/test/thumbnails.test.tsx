import Grid from '@material-ui/core/Grid/Grid'
import { mount, shallow } from 'enzyme'
import React from 'react'
import { act } from 'react-dom/test-utils'
import { defaultViewerState, PreviewImageDataContext, ThumbnailPage, Thumbnails, ViewerStateContext } from '../src'
import { examplePreviewImageData } from './__Mocks__/viewercontext'

/**
 * resize the window for the requested size
 * @param {number} width requested width.
 * @param {number} height requested height.
 */
function resize(width: number, height: number) {
  const resizeEvent = document.createEvent('Event')
  resizeEvent.initEvent('resize', true, true)
  Object.defineProperty(HTMLElement.prototype, 'clientHeight', { configurable: true, value: height })
  Object.defineProperty(HTMLElement.prototype, 'clientWidth', { configurable: true, value: width })
  global.window.dispatchEvent(resizeEvent)
}

describe('Thumbnails component', () => {
  const originalOffsetHeight = Object.getOwnPropertyDescriptor(HTMLElement.prototype, 'offsetHeight')
  const originalOffsetWidth = Object.getOwnPropertyDescriptor(HTMLElement.prototype, 'offsetWidth')
  it('should match snapshot with one page', () => {
    const wrapper = shallow(<Thumbnails onPageClick={() => jest.fn} />)
    expect(wrapper).toMatchSnapshot()
  })

  it('should handle resize', () => {
    const onPageClick = jest.fn()

    Object.defineProperty(HTMLElement.prototype, 'clientHeight', { configurable: true, value: 500 })
    Object.defineProperty(HTMLElement.prototype, 'clientWidth', { configurable: true, value: 500 })

    const wrapper = mount(
      <PreviewImageDataContext.Provider
        value={{
          imageData: [{ ...examplePreviewImageData }, { ...examplePreviewImageData, Index: 2 }],
          setImageData: () => {},
        }}>
        <ViewerStateContext.Provider
          value={{
            ...defaultViewerState,
            showThumbnails: true,
          }}>
          <Thumbnails onPageClick={onPageClick} />
        </ViewerStateContext.Provider>
      </PreviewImageDataContext.Provider>,
    )
    expect(wrapper.find(ThumbnailPage).first().props().viewportHeight).toBe(468)

    act(() => {
      resize(200, 200)
    })

    expect(wrapper.update().find(ThumbnailPage).first().props().viewportHeight).toBe(168)
  })

  it('should handle scroll', () => {
    const onPageClick = jest.fn()

    Object.defineProperty(HTMLElement.prototype, 'clientHeight', { configurable: true, value: 500 })
    Object.defineProperty(HTMLElement.prototype, 'clientWidth', { configurable: true, value: 500 })

    const wrapper = mount(
      <PreviewImageDataContext.Provider
        value={{
          imageData: [{ ...examplePreviewImageData }, { ...examplePreviewImageData, Index: 2 }],
          setImageData: () => {},
        }}>
        <ViewerStateContext.Provider
          value={{
            ...defaultViewerState,
            rotation: [{ degree: 90, pageNum: 1 }],
            activePage: 2,
            showThumbnails: true,
          }}>
          <Thumbnails onPageClick={onPageClick} />
        </ViewerStateContext.Provider>
      </PreviewImageDataContext.Provider>,
    )

    const scrollEvent = document.createEvent('Event')
    scrollEvent.initEvent('scroll', true, true)
    Object.defineProperty(HTMLElement.prototype, 'clientHeight', { configurable: true, value: 100 })
    Object.defineProperty(HTMLElement.prototype, 'clientWidth', { configurable: true, value: 100 })

    act(() => {
      wrapper.find(Grid).getDOMNode().dispatchEvent(scrollEvent)
    })
    expect(wrapper.update().find(ThumbnailPage).first().props().viewportHeight).toBe(68)
  })

  Object.defineProperty(HTMLElement.prototype, 'clientHeight', originalOffsetHeight)
  Object.defineProperty(HTMLElement.prototype, 'clientWidth', originalOffsetWidth)
})
