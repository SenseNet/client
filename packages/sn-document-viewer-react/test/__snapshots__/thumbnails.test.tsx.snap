// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Thumbnails component should match snapshot with one page 1`] = `
<WithStyles(ForwardRef(Grid))
  id="sn-document-viewer-thumbnails"
  innerRef={
    Object {
      "current": undefined,
    }
  }
  item={true}
  style={
    Object {
      "flexGrow": 1,
      "flexShrink": 1,
      "overflow": "auto",
    }
  }
>
  <div
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "column",
        "justifyContent": "center",
        "paddingBottom": 0,
        "paddingTop": 0,
      }
    }
  />
</WithStyles(ForwardRef(Grid))>
`;
